### PR TITLE
ci(code-scanning): add codeql.yml

### PR DESCRIPTION
**Pull request by nitrocode**
## what

- ci(code-scanning): add codeql.yml

## why

- codeql is free, github official, and uses sast and dast to scan the code to find vulnerabilities
- It's used in runatlantis/atlantis and powerful!

## references

- https://github.com/cloudposse/atmos/actions/new?category=security&query=code+scanning
- https://codeql.github.com/
- https://docs.github.com/en/code-security/code-scanning/introduction-to-code-scanning/about-code-scanning-with-codeql

### CODE REVIEW

File: .github/workflows/codeql.yml
Comment (nitrocode) [inline .github/workflows/codeql.yml:1]: This is the file generated from github's codeql

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      matrix:
+        language: [ 'go', 'javascript-typescript' ]
+        # CodeQL supports [ 'c-cpp', 'csharp', 'go', 'java-kotlin', 'javascript-typescript', 'python', 'ruby', 'swift' ]
+        # Use only 'java-kotlin' to analyze code written in Java, Kotlin or both
+        # Use only 'javascript-typescript' to analyze code written in JavaScript, TypeScript or both
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
```
Comment (nitrocode) [inline .github/workflows/codeql.yml:49]:
```suggestion
      uses: actions/checkout@v4
```

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
```
Comment (osterman) [inline .github/workflows/codeql.yml:11]: Let's remove this programmatically generated comment.

Comment (nitrocode) [reply .github/workflows/codeql.yml]: https://github.com/cloudposse/atmos/pull/452/commits/ad2121dd2fee9c735cae7439a37a84a5e303a714

addressed!

**Review by osterman — APPROVED** [review]

**Pull request merged**